### ISSUE TITLE
Delay does not seems to be tacken into account

### ISSUE THREAD

**Issue by AlexandreRoba** [avast/retry-go#5]
Hi,
I have the following code:
```Go
var eventPublisher *eventpublisher.EventPublisher
	err = retry.Do(func() error {
		eventPublisher, err = eventpublisher.NewEventPublisher(amqpURL, *amqpEventsExchange)
		if err != nil {
			return err
		}
		return nil
	},
		retry.Delay(5000*time.Second),
		retry.Attempts(30),
	)
```

I can indeed see 30 retries in the log but they all happen in the same second. Any idea?

Thks,
Alex

**Comment by JaSei** [comment]
Hi, thank you for your report. It's bug and will be fix today (#6)...

**Comment by JaSei** [comment]
fixed and released as 1.0.1

**Comment by AlexandreRoba** [comment]
Excellent, I'll give it a try. thanks for the swift bug correction.
